ext/json/json_parser.y

```diff
@@ -316,7 +358,7 @@ PHP_JSON_API void php_json_parser_init(php_json_parser *parser,
 			str_len,
 			options,
 			max_depth,
-			&default_parser_methods);
+			((options & PHP_JSON_VALIDATE_ONLY) ? &validate_parser_methods : &default_parser_methods));
```

Comment:
I did it like that because I was following the pattern used by json_decode().
But I think is a good idea your proposal, let me check it to see if I can get it done correctly

Reply:
@bukka I already pushed a change about this